File: beginner_tutorials/scripts/listener.py
#!/usr/bin/env python      
# -*- coding: utf-8 -*-

import rospy
from beginner_tutorials.msg import Song

def callback(data):
    print ('name:%s',data.name)
    print ('singger:%s',data.singger)
    print ('%s',data.lyrics)
    print ("")

def listener():
    rospy.init_node('listener', anonymous = False)
    sub = rospy.Subscriber("chatter", Song, callback)
    rospy.spin()


if __name__ == '__main__':
    listener()

File: beginner_tutorials/src/listener.cpp
#include "ros/ros.h"
#include "beginner_tutorials/Song.h"
#include <iostream>
#include <string>

void callback(const beginner_tutorials::SongConstPtr& msg)
{
	std::cout << "name:" << msg->name << std::endl
		      << "singger:" << msg->singger << std::endl
    		  << msg->lyrics << std::endl;
}

int main(int argc, char *argv[])
{
	ros::init(argc, argv, "listener");
	ros::NodeHandle n;
	ros::Subscriber sub = n.subscribe("chatter", 10, callback);
	ros::spin();

	return 0;
}